### ISSUE TITLE
Add unsupported PHP version?

### ISSUE THREAD

**Issue by dakira** [laravel/homestead#727]
It would be great to have at least one older (unsupported) PHP version, like 5.3 or 5.4.

### Rationale
It happens from time to time that you need to work on a legacy app. For example to upgrade it to a point where you can use it with supported PHP versions. This work is currently not (easily) possible within Homestead.

### Proposals
- Add a manually compiled PHP 5.4 to the Homestead box (and adjust the scripts accordingly)
- Add a script that allows users to specify the PHP version they need, download and compile the source and set everything up.

### Implications
As it's a local development box, I would say none. People don't use Homestead to host their sites.

What do you think?

**Comment by svpernova09** [comment]
Thanks for taking the time to open the issue.

While I completely understand the desire for supporting old versions I think that it's something much easier said than done. If we did go back and support older versions we wouldn't compile from source. We'd just use older versions from the available repos. I don't ever want Homestead to have to compile PHP from source if we can ever avoid it.

Another concern in doing this is if you are running on 5.3 or 5.4 you're likely not running MySQL 5.7 (or the current Postgres version) that we install by default. Once you get to the point where you're having to downgrade and support multiple versions of more and more applications it just becomes an exponentially increasing configuration problem. Since Homestead has always been a simple tool (Using shell scripts VS Chef / Puppet / Ansible) I don't want to have to get into the business of building all those various combinations and then having to support them.

Homestead supports back to 5.6 because that's what we can easily do via [Ondřej's repo](https://launchpad.net/~ondrej/+archive/ubuntu/php). If older versions than 5.6 were there I'd happily add them. I believe he's dropped support there because they're EOL (or close). As soon as Ondřej stops supporting 5.6 there, we will likely stop supporting it as well.

TL;DR - no plans for this because I don't want my brain to explode :D

**Comment by dakira** [comment]
I totally understand. Ondrej's repos are being used a lot on production servers which is why he stopped "supporting" EOL PHP versions.

In development envrionments it's something completely different IMO. You sometimes need EOL PHP - if only for a minute - to upgrade something to support never PHP versions.

I think the route with the script wouldn't be as intrusive as adding a manially compiled PHP to the container. If people really want PHP 5.2 in their box they can run enable_eol_php.sh 5.2, wait 10 minutes and use it as they would with 5.6 or 7.0. We could probably even get the build scripts from Ondrej to get the same setup as with the offically supported versions.

**Comment by dakira** [comment]
@svpernova09 oh.. and I forgot.. I would do the work, of course. ;-) I was just testing the water if a PR would be accepted.

**Comment by svpernova09** [comment]
I truly welcome any and all PRs. But won't guarantee they'll be merged. But I will be happy to provide feedback.

My main concern here is not adding the feature. It's the ongoing support I'd have to do for it. 

That being said: If you're so inclined feel free to open a PR and we can see what it looks like.

**Comment by dakira** [comment]
>  It's the ongoing support I'd have to do for it.
Honestly.. as those PHP versions are EOL.. there is no support. ;) Of course a script would have to be changed in case Ondrej changes up his PHP PPA forcing all PHP stuff in Homestead to change. Anyhow.. I'll go the script route and see what I can whip up.
